### DISCUSSION TITLE
Is it possible to make web app installable from a dynamic route?

### DISCUSSION THREAD

**Discussion by heecheon92** [serwist/serwist#96] [Q&A]
Let's say there is a page with dynamic route where a user can only enter under completing certain condition.

For example, "localhost:3000/[lang]/uid=blahblah"

The actual URL user would lane would be something like. "localhost:3000/en/uid=abcdefg"

Once, the user lands on the url, I want our web to be installable with start_url = "localhost:3000/en/uid=abcdefg"

Which means, each user will have the same app installed starting from different start_url.

Is such a behavior possible?

**Comment by DuCanhGH** [comment]
@heecheon92 you can add a middleware that redirects the user from /\<path\> to /[lang]/\<path\>. Not so sure about having search params in `start_url`, but I'd say that a dynamically generated manifest.json with the desired `start_url` is probably your best bet in that case.
